### PR TITLE
Dockerise oag

### PR DESCRIPTION
**Pull request by david-wirelab**
YEL-1513

### CODE REVIEW

File: ADT/sftp_oag_client_maytech.py [hunk 1]
```diff
@@ -41,64 +48,59 @@ def ssh_login(in_host, in_user, in_keyfile):
...
-        return False
+    file_list = os.listdir(filename)
+    for scan_file in file_list:
+        processing = os.path.join(STAGING_DIR,scan_file)
+        with open(processing, 'rb') as scan:
+          response = requests.post('http://' + BASE_URL + ':' + BASE_PORT + '/scan', files={'file': scan}, data={'name': scan_file})
+        if not 'Everything ok : true' in response.text:
+          logger.error('File %s is dangerous, preventing upload' % filename)
+          file_quarantine = os.path.join(QUARANTINE_DIR,scan_file)
+          logger.info('Move %s from staging to quarantine %s',processing ,file_quarantine)
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py]: Linting-wise, we should have whitespace after the commas, not before.

File: ADT/sftp_oag_client_maytech.py [hunk 2]
```diff
@@ -41,64 +48,59 @@ def ssh_login(in_host, in_user, in_keyfile):
...
+          os.rename(processing,file_quarantine)
+          return False
     else:
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py]: This `else` is now incorrectly indented as the original `if` statement it was attached to has been replaced with the `for` loop. It needs to be indented in order to attach to the `if not ...`

File: ADT/sftp_oag_client_maytech.py [hunk 3]
```diff
@@ -41,64 +48,59 @@ def ssh_login(in_host, in_user, in_keyfile):
...
-            format="%(asctime)s\t%(name)s\t%(levelname)s\t%(message)s",
-            datefmt='%Y-%m-%d %H:%M:%S',
-            level=logging.INFO
-        )
-
+    # Logging to Docker stdout
     logger = logging.getLogger()
-    logger.info("Starting")
-    status = 1
+    logger.setLevel(logging.INFO)
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py]: The log level should be set via environment variable.

File: ADT/sftp_oag_client_maytech.py [hunk 4]
```diff
@@ -41,64 +48,59 @@ def ssh_login(in_host, in_user, in_keyfile):
...
+
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.INFO)
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py]: Please set the log level via an environment variable

File: ADT/sftp_oag_client_maytech.py [hunk 5]
```diff
@@ -82,23 +95,24 @@ def main():
...
-    logger.debug("Connecting via SSH")
-    ssh = ssh_login(ssh_remote_host_maytech, ssh_remote_user_maytech, ssh_private_key)
-    logger.debug("Connected")
+    uploadcount = 0
+    logger.info("Connecting via SSH")
+    ssh = ssh_login(SSH_REMOTE_HOST_MAYTECH, SSH_REMOTE_USER_MAYTECH, SSH_PRIVATE_KEY)
+    logger.info("Connected")
     sftp = ssh.open_sftp()
 
     try:
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py:114]: This `try` has no `except` to catch any exceptions

File: ADT/sftp_oag_client_maytech.py [hunk 6]
```diff
@@ -151,11 +164,27 @@ def main():
...
-        status=-1
 
-    logger.info("Done Status %s", status)
-    print status
+# Move files to S3
+    logger.info("Starting to move files to S3")
+    processed_oag_file_list = [f for f in os.listdir(DOWNLOAD_DIR)]
+    if processed_oag_file_list:
+        for filename in processed_oag_file_list:
+            boto_s3_session = boto3.Session(
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py]: Creating a new boto session per file would seem inefficient, versus starting the session before the `for` loop.

**Review by tomfitzherbert — CHANGES_REQUESTED** [review]

File: ADT/sftp_oag_client_maytech.py (hunk 5 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py:114]: It's in Line 147

File: ADT/sftp_oag_client_maytech.py (hunk 1 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py]: Corrected.

File: ADT/sftp_oag_client_maytech.py (hunk 2 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py]: Indented

File: ADT/sftp_oag_client_maytech.py (hunk 3 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py]: Reverted change.

File: ADT/sftp_oag_client_maytech.py (hunk 4 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py]: Reverted change.

File: ADT/sftp_oag_client_maytech.py (hunk 5 above)
Comment (tomfitzherbert) [reply ADT/sftp_oag_client_maytech.py:114]: Ah! GitHub had hidden/collapsed that part

File: ADT/sftp_oag_client_maytech.py (hunk 6 above)
Comment (david-wirelab) [reply ADT/sftp_oag_client_maytech.py]: Corrected.

File: ADT/sftp_oag_client_maytech.py [hunk 7]
```diff
@@ -132,30 +146,44 @@ def main():
...
+    logger.info("Starting to move files to S3")
+    processed_oag_file_list = [f for f in os.listdir(DOWNLOAD_DIR)]
+    if processed_oag_file_list:
+        boto_s3_session = boto3.Session(
+            aws_access_key_id=S3_ACCESS_KEY_ID,
+            aws_secret_access_key=S3_SECRET_ACCESS_KEY,
+            region_name=S3_REGION_NAME
+        )
+        for filename in processed_oag_file_list:
+            s3 = boto_s3_session.client("s3")
```
Comment (tomfitzherbert) [inline ADT/sftp_oag_client_maytech.py:178]: Sorry @david-wirelab, I should have been clearer; it's probably worth moving the s3 client creation to outside the `for` loop too.

**Review by tomfitzherbert — CHANGES_REQUESTED** [review]

**Comment by david-wirelab** [comment]
Closing this PR as the actual script has been moved to https://github.com/UKHomeOffice/dq-oag-sftp-python/tree/use-new-script

**Pull request closed without merging**